telp-dashboard-v1/auth.config.ts

```diff
@@ -10,9 +10,17 @@ export const authConfig = {
   callbacks: {
     authorized({ auth, request: { nextUrl } }) {
       const isLoggedIn = !!auth?.user;
+      const isNotAdmin = false // Check if user.role === 'teacher';
       const isOnDashboard = nextUrl.pathname.startsWith('/dashboard');
+      const isOnAdminLevelPage = (
+        nextUrl.pathname.startsWith('/dashboard/teachers') || 
+        nextUrl.pathname.startsWith('/dashboard/admins')
+      );
       if (isOnDashboard) {
-        if (isLoggedIn) return true;
+        if (isLoggedIn) {
+          if (isOnAdminLevelPage && isNotAdmin) return Response.redirect(new URL('/dashboard', nextUrl)); // Redirect teachers to dashboard
```

Comment:
Isn't this already taken care of by the cookies? Our role-checking logic uses the cookies to make sure non-Admins cannot even view an AdminLevelPage. This would make this portion of code redundant..would it not?

Reply:
Ah, I forgot to change that back after I was messing with it. It works the same way as we had it before currently, but I'll make sure to change it back tomorrow once the pull goes through.